packages/polling-controller/src/StaticIntervalPollingController.test.ts

```diff
@@ -48,15 +53,18 @@ describe('StaticIntervalPollingController', () => {
     controller.setIntervalLength(TICK_TIME);
     clock = useFakeTimers();
   });
+
   afterEach(() => {
     clock.restore();
+    controller.executePollPromises = []
```

Comment:
this may be a telltale sign that we should create a fresh controller instance for each scenario